client/app/core/templates.service.js

```diff
@@ -0,0 +1,76 @@
+/* eslint camelcase: "off" */
+
+/** @ngInject */
+export function TemplatesServiceFactory(CollectionsApi) {
+  const collection = 'orchestration_templates';
+  const service = {
+    getMinimal: getMinimal,
+    getTemplates: getTemplates,
+  };
+
+  return service;
+
+  function getMinimal(filters) {
+    const options = {
+      filter: getQueryFilters(filters),
+      hide: 'resources',
+    };
+
+    return CollectionsApi.query(collection, options);
+  }
+
+  function getTemplates(limit, offset, filters, sorting) {
+    const options = {
+      expand: ['resources'],
+      limit: limit,
+      offset: String(offset),
+      attributes: [],
+      filter: getQueryFilters(filters),
+    };
+
+    if (angular.isDefined(sorting)) {
+      options.sort_by = sorting.field;
+      options.sort_options = sorting.sortOptions;
+      options.sort_order = sorting.direction;
+    }
+
+    return CollectionsApi.query(collection, options);
+  }
+
+  // Private
+
+  function getQueryFilters(filters) {
+    const queryFilters = [];
+
+    angular.forEach(filters, function (nextFilter) {
+      if (nextFilter.id === 'name') {
+        queryFilters.push("name='%" + nextFilter.value + "%'");
```

Comment:
Are you talking about the code below this section that is doing regular expression matching?   The logic needed to be written because the orchestration template type names are different than what a person would try and type to filter on.  Partial matches wouldn't even work in this case.  For example, OrchestrationTemplateCfn is the template type for AWS cloudformation scripts.  I had a talk with @chriskacerguis about this and this seemed liked the least intrusive solution we came up with.